### PR TITLE
fix: IoTDB bridge incoming payload needs to be parsed as JSON

### PR DESCRIPTION
**Pull request by kjellwinblad**
There was an incorrect assumption that the data incoming to the IoTDB bridge has already been parsed. This is fixed by parsing the payload as JSON data if the payload is not already a map.

Fixes:
https://emqx.atlassian.net/browse/EMQX-9854

### <samp>🤖 Generated by Copilot at a885e0c</samp>

Refactored the `preproc_data` function in `emqx_bridge_iotdb_impl.erl` to support different data formats and improved readability. Bumped the version number of `emqx_bridge_iotdb` to 0.1.1.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [] Added tests for the changes
- [] Changed lines covered in coverage report
- [] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [] Schema changes are backward compatible

### CODE REVIEW

**Comment by kjellwinblad** [comment]
I was too fast. Found more issues and will try to fix them in this PR as well.

**Comment by kjellwinblad** [comment]
Made a better fix. This is ready for review now.

File: apps/emqx_bridge_iotdb/src/emqx_bridge_iotdb_impl.erl [hunk 1]
```diff
@@ -258,12 +276,15 @@ convert_float(Str) when is_binary(Str) ->
 convert_float(undefined) ->
     null.
 
-make_iotdb_insert_request(Message, State) ->
+make_iotdb_insert_request(MessageUnparsedPayload, State) ->
+    PayloadUnparsed = maps:get(payload, MessageUnparsedPayload),
+    PayloadParsed = make_parsed_payload(PayloadUnparsed),
+    Message = MessageUnparsedPayload#{payload => PayloadParsed},
```
Comment (thalesmg) [inline apps/emqx_bridge_iotdb/src/emqx_bridge_iotdb_impl.erl:282]: Tiny nit:

```suggestion
    Message = maps:update_with(payload, fun make_parsed_payload/1, MessageUnparsedPayload),
```

**Review by thalesmg — DISMISSED** [review]

**Comment by kjellwinblad** [comment]
@thalesmg committed your suggestion so will need another approval. Can you also help me merge if you can as I don't have permission.

**Review by thalesmg — APPROVED** [review]

**Pull request merged**